apps/expo/src/components/Discussion.tsx

```diff
@@ -38,6 +38,50 @@ export default function Discussion({
   const [showCommentInput, setShowCommentInput] = useState(false);
   const [commentText, setCommentText] = useState("");
 
+  const [isLiked, setIsLiked] = useState(false);
+  const [likesCount, setLikesCount] = useState(0);
+
+  const { data: likesData } = api.like.getLikesForDiscussion.useQuery({
+    postId: discussion.id,
+  });
+
+  useEffect(() => {
```

Comment:
Do you mind explaining the purpose of this `useEffect` hook? It's not immediately clear to all renders. Additionally, since your `likesData` is your dependency, isn't this block only run on the first fetch of the `likesData`? Was this intentional?